### PR TITLE
[Feature][Connector-V2] Jdbc support trino dialect

### PR DESCRIPTION
**Pull request by sanjaychitransh**
### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Thanks @sanjaychitransh ! Please fix ci.

**Comment by sanjay-ibm** [comment]
> Thanks @sanjaychitransh ! Please fix ci.

Hi @Hisoka-X, thanks ! for the review, I fixed CI issues,  but seems one build is failing (engine-V2-it),but not due to code issue

**Comment by Hisoka-X** [comment]
> > Thanks @sanjaychitransh ! Please fix ci.
> 
> Hi @Hisoka-X, thanks ! for the review, I fixed CI issues, but seems one build is failing (engine-V2-it),but not due to code issue

You can just retry failed ci. No need re-trigger all.

File: seatunnel-connectors-v2/connector-jdbc/pom.xml [hunk 1]
```diff
@@ -368,6 +379,14 @@
...
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
+                </configuration>
+            </plugin>
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/pom.xml]: Please revert

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/trino/TrinoTypeMapper.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            case TRINO_ARRAY:
+            case TRINO_ROW:
+            default:
+                final String jdbcColumnName = metadata.getColumnName(colIndex);
+                throw new JdbcConnectorException(
+                        CommonErrorCode.UNSUPPORTED_OPERATION,
+                        String.format(
+                                "Doesn't support TRINO type '%s' on column '%s'  yet.",
+                                columnType, jdbcColumnName));
+        }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/trino/TrinoTypeMapper.java]: Please create common error exception through https://github.com/apache/seatunnel/blob/392300ea023f3879c84a9b98a61da19a0e31ceb0/seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonError.java#L58

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/pom.xml [hunk 3]
```diff
@@ -90,5 +95,93 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-e2e-common</artifactId>
+            <version>2.3.9-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-e2e-common</artifactId>
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/pom.xml]: why?

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcTrinoIT.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static org.awaitility.Awaitility.given;
+
+@Slf4j
+public class JdbcTrinoIT extends TestSuiteBase implements TestResource {
+
+    private static final String DOCKER_IMAGE = "trinodb/trino";
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcTrinoIT.java]: Let's fix the trino version to make sure the test case stable?

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_trino_source_and_console .conf [hunk 5]
```diff
@@ -0,0 +1,48 @@
+#
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_trino_source_and_console .conf]: I can't find the place to trigger this test case.

File: seatunnel-connectors-v2/connector-jdbc/pom.xml (hunk 1 above)
Comment (sanjaychitransh) [reply seatunnel-connectors-v2/connector-jdbc/pom.xml]: reverted

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/pom.xml (hunk 3 above)
Comment (sanjaychitransh) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/pom.xml]: removed

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcTrinoIT.java (hunk 4 above)
Comment (sanjaychitransh) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcTrinoIT.java]: added trino version for test

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/trino/TrinoTypeMapper.java (hunk 2 above)
Comment (sanjaychitransh) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/trino/TrinoTypeMapper.java]: did the same way

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_trino_source_and_console .conf (hunk 5 above)
Comment (sanjaychitransh) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_trino_source_and_console .conf]: remove unused file

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcTrinoIT.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            dbServer.close();
+        }
+    }
+
+    @TestTemplate
+    @DisplayName("JDBC-Trino end to end test")
+    public void testJdbcTrino(TestContainer container) {
+        assertHasData();
+        log.info(container.toString());
+    }
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcTrinoIT.java]: I can not find the test case of trino. Please add a test case for it. You can refer https://github.com/apache/seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-6/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcHanaIT.java#L105

**Comment by krutoileshii** [comment]
so is this merged? or needs to be redone ?

**Comment by sanjaychitransh** [comment]
its merged

**Pull request closed without merging**